spec/lib/efile/nj/nj_flat_eitc_eligibility_spec.rb

```diff
@@ -0,0 +1,333 @@
+              .and_return(test_case[:meets_age_requirements])
+          end
+
+          unless test_case[:is_under_income_total_limit].nil?
+            allow(Efile::Nj::NjFlatEitcEligibility)
+              .to receive(:is_under_income_total_limit?)
+              .and_return(test_case[:is_under_income_total_limit])
+          end
+
+          unless test_case[:has_ssn_valid_for_employment].nil?
+            allow(Efile::Nj::NjFlatEitcEligibility)
+              .to receive(:has_ssn_valid_for_employment?)
+              .and_return(test_case[:has_ssn_valid_for_employment])
+          end
+
+          unless test_case[:claimed_as_dependent].nil?
+            allow(Efile::Nj::NjFlatEitcEligibility)
+              .to receive(:claimed_as_dependent?)
+              .and_return(test_case[:claimed_as_dependent])
+          end
+          
+          result = Efile::Nj::NjFlatEitcEligibility.possibly_eligible?(intake)
+          expect(result).to eq(test_case[:expected])
+        end
+      end
+    end
+  end
+
+  describe ".investment_income_over_limit?" do
+    context "when above limit" do
+      let(:intake) { create(:state_file_nj_intake, :df_data_investment_income_12k) }
+      it "returns true" do
+        expect(Efile::Nj::NjFlatEitcEligibility.investment_income_over_limit?(intake)).to eq(true)
+      end
+    end
+
+    context "when under limit" do
+      let(:intake) { create(:state_file_nj_intake, :df_data_minimal) }
+      it "returns false" do
+        expect(Efile::Nj::NjFlatEitcEligibility.investment_income_over_limit?(intake)).to eq(false)
+      end
+    end
+  end
+
+  describe ".meets_age_requirements?" do
+    current_tax_year = MultiTenantService.statefile.current_tax_year
+    year_65 = current_tax_year - 65 # for 2024: 1959
+    year_25 = current_tax_year - 25 # for 2024: 1999
+    year_18 = current_tax_year - 18 # for 2024: 2006
+
+    date_older_than_65 = Date.new(year_65+1, 1, 1) # 1/1/1960
+    date_younger_than_65 = Date.new(year_65+1, 1, 2) # 1/2/1960
+
+    date_older_than_25 = Date.new(year_25+1, 1, 1) # 1/1/2000
+    date_younger_than_25 = Date.new(year_25+1, 1, 2) # 1/2/2000
+
+    date_older_than_18 = Date.new(year_18, 12, 31) # 12/31/2006
+    date_younger_than_18 = Date.new(year_18+1, 1, 1) # 1/1/2007
+
+    context "when mfj" do
+      [
+        # 2024: Taxpayer is born on or before 1/1/1960. If MFJ, both taxpayers have to meet the requirement.
+        { primary_birth_date: date_older_than_65, spouse_birth_date: date_older_than_65, expected: true },
+        { primary_birth_date: date_younger_than_65, spouse_birth_date: date_older_than_65, expected: false },
+        { primary_birth_date: date_older_than_65, spouse_birth_date: date_younger_than_65, expected: false },
+        { primary_birth_date: date_younger_than_65, spouse_birth_date: date_younger_than_65, expected: false },
+
+        # 2024: Taxpayer is born on or after 1/1/2000. If MFJ, both taxpayers have to meet the requirement.
```

Comment:
@mluedke2 is this comment out of date? Having trouble parsing the code but I see # 1/2/2000 in meets_age_requirements above (which is correct) so my hunch is the implementation is right

Reply:
you're right, comments here are stale-- didn't change when i did the latest code update. good catch, will update these

Reply:
fixed